### DISCUSSION TITLE
-Ykind-projector does not support infix type

### DISCUSSION THREAD

**Discussion by xuwei-k** [scala/scala3#17896] [Feature Requests]
## Minimized code

dotty version: `0.24.0-RC1`, `0.25.0-bin-20200508-f5b1d79-NIGHTLY`

```Scala
scala> trait Functor[F[_]]
// defined trait Functor

scala> type EitherFunctor[A] = Functor[Either[A, *]]
// defined alias type EitherFunctor[A] = Functor[[_$1] =>> Either[A, _$1]]

scala> type EitherFunctor[A] = Functor[A Either *]
1 |type EitherFunctor[A] = Functor[A Either *]
  |                                         ^
  |                                         Not found: type *

```

## Expectation

compile success

**Comment by smarter** [comment]
We provide `-Ykind-projector` to help with migration, but we don't plan to extend it ourselves to support more of its features. PRs to do so are welcome though, so I'll move this issue to the feature request repo.

**Comment by xuwei-k** [comment]
FYI: created scalafix rule

https://github.com/xuwei-k/scalafix-rules/commit/cde493f8d75fa46d5afb754dfa2d6dc9a842a6b8

**Comment by TomasMikula** [comment]
From [Kind Projector Migration](https://docs.scala-lang.org/scala3/guides/migration/plugin-kind-projector.html) page:

> In the future, Scala 3 will use the `_` underscore symbol for placeholders in type lambdas—just as the underscore is currently used for placeholders in (ordinary) term-level lambdas.

The above was written 4+ years ago, so I'm wondering when the "future" will arrive. Is the feature planned for a specific Scala version?
